src/vos/vos_io.c

```diff
@@ -881,8 +889,20 @@ akey_update(struct vos_io_context *ioc, uint32_t pm_ver, daos_handle_t ak_toh)
 	} /* else: array */
 
 	for (i = 0; i < iod->iod_nr; i++) {
+		umem_off_t	umoff = iod_update_umoff(ioc);
```

Comment:
shouldn't we not call this in akey_update_recx if we call it here?

Reply:
If we want to call iod_update_umoff() inside akey_update_recx(), then we also need to check "rx_nr == 0" inside akey_update_recx() to avoid leak iod_update_umoff(). Currently, both iod_update_umoff() and "rx_nr == 0" are handled outside of akey_update_recx(), so it works.

Reply:
Ah, I see.   So this is just for the assertion.  It's unused by akey_update_recx (I was looking at akey_update_single).  I really miss gerrit's ability to expand the whole file (rather than github's click to expand 10 lines at a time)